### PR TITLE
mgmt: mcumgr: callback: Add op to receive callback

### PR DESCRIPTION
**Pull request by thedjnK**
Adds the operation supplied by the client to the MCUmgr command received callback

### CODE REVIEW

File: subsys/mgmt/mcumgr/smp/src/smp.c [hunk 1]
```diff
@@ -203,7 +203,7 @@ static int smp_handle_single_payload(struct smp_streamer *cbuf, const struct smp
 #if defined(CONFIG_MCUMGR_SMP_COMMAND_STATUS_HOOKS)
 		cmd_recv.group = req_hdr->nh_group;
 		cmd_recv.id = req_hdr->nh_id;
-		cmd_recv.err = MGMT_ERR_EOK;
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/smp/src/smp.c:206]: Shouldn't this be part of another commit? It does not seem to be explained by commit description.

Comment (nordicjm) [reply subsys/mgmt/mcumgr/smp/src/smp.c:206]: It was repurposed because it shouldn't have been there as per `/** #mcumgr_err_t, used in #MGMT_EVT_OP_CMD_DONE */`

**Review by de-nordic — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by nordicjm** [comment]
> I supposedly should not merge this without assignee approval. :-)

Can't approve my own commits :)

**Comment by fabiobaltieri** [comment]
> Can't approve my own commits :)

We are stuck in an unmergiability loophole!
